bigbite/themer
# [BBT-101] Filter UI component

## Description



Completes [BBT-101](https://b5ecom.atlassian.net/browse/BBT-101) - This PR adds a filter UI component, which allows you to set a duotone filter on specific blocks. 

This property in `theme.json` requires you to specify a css variable, so I've disabled all ability to choose custom colours on the `DuotonePicker` component. A part of this had to be done using CSS as there was no prop to control its visibility.

## Change Log



- Added filter UI component.

## Steps to test



- Add an image to the preview on your local setup e.g.
```js
createBlock( 'core/image', {
    id: 1730,
    sizeSlug: 'full',
    linkDestination: 'none',
    url: 'https://blocks-playground.bigbite.site/wp-content/uploads/2022/11/fish.webp',
} )
```
- Add a filter property for the image block in `theme.json`
```json
"styles": {
  "core/image": {
    "filter":  {
          "duotone": "var(--wp--preset--duotone--blue-red)"
        }
  },
```
- Check the correct duotone option is selected based on your theme.json when you first load in.
- Choose another duotone option from the list and check it is applied to the image. I can't see any visible change in the preview here, but I can see the CSS being applied in the console.
- Export `theme.json` and check it exports with the correct var name.

## Screenshots/Videos



### Importing

https://github.com/bigbite/themer/assets/45216035/47e859b2-4d0b-4b66-a5e7-aa6aa165fbac

### Changing value

https://github.com/bigbite/themer/assets/45216035/2d469ed3-5120-43a6-b17f-c0c8adba14ff

### Exporting

https://github.com/bigbite/themer/assets/45216035/abe34e23-0ad8-43ff-8da1-000bfaa03b0d

## Checklist:





[BBT-101]: https://b5ecom.atlassian.net/browse/BBT-101?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
